File: 프로그래머스/lv2/70129. 이진 변환 반복하기/이진 변환 반복하기.js
function solution(s) {
    const answer = [];
    
    let countTrans = 0;
    let countZero = 0;
    
    while (s.length !== 1) {
        countTrans++;
        countZero += s.split("").filter((item) => item === '0').length;
        s = s.replace(/0/g, "")
        s= s.length.toString(2);
    }
    
    answer.push(countTrans);
    answer.push(countZero);
    
    return answer;
}

File: 프로그래머스/lv2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    count = 0
    removed_zero = 0
    while True:
        count += 1
        removed_zero += s.count('0')
        
        s = s.replace('0', '')
        
        if s == '1':
            break
        
        c = len(s)
        
        tmp = ''
        while c:
            n, m = divmod(c, 2)
            c = n
            tmp += str(m)
        
        s = tmp[::-1]
        
    return [count, removed_zero]